jira_agile_metrics/calculators/cycletime.py

```diff
@@ -237,25 +235,28 @@ def calculate_cycle_times(
                 impediment_flag = None
                 impediment_start = None
                 impediment_start_status = None
-            
-            # Wipe timestamps if items have moved backwards; calculate cycle time
+
+            # calculate cycle time
 
             previous_timestamp = None
-            accepted_timestamp = None
-            completed_timestamp = None
+            committed_timestamp = None
+            done_timestamp = None
 
-            for cycle_name in cycle_names:
+            for cycle_name in reversed(cycle_names):
```

Comment:
Just so that I can make sure I am understanding this correctly, could you explain the thinking behind doing this check in reverse order and what the implications are? Does this fix your bug?

Reply:
When a column gets skipped, I use the next timestamp to fill in in skipped columns. For example if I have a workflow `A`, `B`, `C` and an issue goes from `A` directly to `C`, the I use the same timestamp from `C` for `B`. This is semantically correct and makes calculations much simpler. Semantically it means that an issue spent zero seconds in `B` before it moves on to `C`. ActionableAgile Analytics does it the same way.